### ISSUE TITLE
define median-based error metric for keras in R

### ISSUE THREAD

**Issue by msloryg** [rstudio/keras#780]
I would like to use the custom metric MdAPE (median absolute percentage error), but since I could neither find a median backend function (something like k_median) nor something like "order" or "sort" (so I could calculate the median myself) I do not really know how to do it..
I think median-based error metrics are not uncommon so I am quite surprised I couldn't find anything.

**Comment by skeydan** [comment]
I think you could define a custom metric 

https://tensorflow.rstudio.com/keras/reference/metric_binary_accuracy.html

and use `percentile` from TensorFlow Probability to compute the quantiles.

If you install `tfprobability`

https://rstudio.github.io/tfprobability/

, you can use this as

`tfp$stats$percentile`

**Comment by msloryg** [comment]
Thanks for the tip! I installed tfprobability but unfortunately R says it cannot find it... This is what I did:
```
devtools::install_github("rstudio/tfprobability")
library(tfprobability)
library(tensorflow)
```
I actually already have tensorflow (and use it) but I reinstalled it just to be sure:
```
install_tensorflow()
tfe_enable_eager_execution() #this was recommended on the link 
```
Then I tried the example on the website:
```
d <- tfd_binomial(total_count = 7, probs = 0.3)
Error: Python module tensorflow_probability was not found.

Detected Python configuration:

python:         C:\Users\Laura\AppData\Local\Programs\Python\Python36\python.exe
libpython:      C:/Users/Laura/AppData/Local/Programs/Python/Python36/python36.dll
pythonhome:     C:\Users\Laura\AppData\Local\Programs\Python\Python36
version:        3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:54:40) [MSC v.1900 64 bit (AMD64)]
Architecture:   64bit
numpy:          C:\Users\Laura\AppData\Local\Programs\Python\Python36\lib\site-packages\numpy
numpy_version:  1.16.1
tensorflow_probability:[NOT FOUND]

python versions found: 
 C:\Users\Laura\Anaconda3\envs\r-tensorflow\python.exe
 C:\Users\Laura\AppData\Local\Programs\Python\Python36\python.exe
 C:\Users\Laura\AppData\Local\Programs\Python\Python36\\python.exe
 C:\Users\Laura\ANACON~1\python.exe
 C:\Users\Laura\Anaconda3\python.exe
 C:\Users\Laura\Anaconda3\envs\tensorflow_env\python.exe
```
Do you know what the issue could be?

**Comment by skeydan** [comment]
can you try the nightly versions

```
tensorflow:::install_tensorflow(version = "nightly")
```

and see if 

```
library(tensorflow)
library(tfprobability)
```

works?
